WordPress/src/main/java/org/wordpress/android/ui/domains/management/newdomainsearch/domainsfetcher/NewDomainsSearchRepository.kt

```diff
@@ -0,0 +1,72 @@
+package org.wordpress.android.ui.domains.management.newdomainsearch.domainsfetcher
+
+import org.wordpress.android.Constants
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.SiteActionBuilder
+import org.wordpress.android.fluxc.model.products.Product
+import org.wordpress.android.fluxc.store.ProductsStore
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.SiteStore.OnSuggestedDomains
+import org.wordpress.android.util.dispatchAndAwait
+import javax.inject.Inject
+
+private const val SUGGESTIONS_REQUEST_COUNT = 20
+
+class NewDomainsSearchRepository @Inject constructor(
+    private val productsStore: ProductsStore,
+    private val dispatcher: Dispatcher
+) {
+    var products: List<Product>? = null
+
+    suspend fun searchForDomains(query: String): DomainsResult {
+        if (products == null) fetchProducts()
+        return SiteActionBuilder.newSuggestDomainsAction(
+            SiteStore.SuggestDomainsPayload(
+                query = query,
+                onlyWordpressCom = false,
+                includeWordpressCom = false,
+                includeDotBlogSubdomain = false,
+                quantity = SUGGESTIONS_REQUEST_COUNT
+            )
+        ).let { action ->
+            dispatcher.dispatchAndAwait<SiteStore.SuggestDomainsPayload, OnSuggestedDomains>(action)
+        }.let { event ->
+            onDomainSuggestionsFetched(query, event)
+        }
+    }
+
+    private suspend fun fetchProducts() {
+        val result = productsStore.fetchProducts(Constants.TYPE_DOMAINS_PRODUCT)
+        if (!result.isError) result.products?.let { products = it }
+    }
+
+    private fun onDomainSuggestionsFetched(query: String, event: OnSuggestedDomains): DomainsResult {
+        return if (query == event.query && !event.isError) {
+            val suggestions = event.suggestions
+                .filter { !it.is_free }
+                .sortedBy { it.relevance }
+                .map { domain ->
+                    val product = products?.firstOrNull { product -> product.productId == domain.product_id }
+                    val splitDomainName = domain.domain_name.split('.')
+                    val suffix = splitDomainName.last()
+                    val prefix = domain.domain_name.removeSuffix(suffix)
```

Comment:
Not suggesting to change this, but I'm curious if we used:

```suggestion
                    val prefix = splitDomainName.dropLast(1).joinToString("")
```

Is there a difference?

Reply:
A little bit. With this approach, we lose `.`

Reply:
Ah, right, and even with `joinToString(".")`, there'd be a missing '.' at the end. Thanks for helping me reason about the logic. 😄